src/Solvers/preconditioned_conjugate_gradient_solver.jl

```diff
@@ -172,9 +172,9 @@ function solve!(x, solver::PreconditionedConjugateGradientSolver, b, args...)
         iterate!(x, solver, b, args...)
     end
 
-    fill_halo_regions!(x, solver.architecture)
+    fill_halo_regions!(x, solver.architecture) # blocking
```

Comment:
what does # blocking mean?

Reply:
By that I mean there is a `wait` inside which prevents computation from continuing until all the tasks / kernels launched inside the function are complete.

Though its tempting to think about the code executing in serial, in reality the code can execute asynchronously and we have to manually control when we stop and wait for a computation / task to finish. So the comment here is a little reminder that this line will block until its done, unlike some other functions.

Reply:
Thanks!